### PR TITLE
Advanced configuration dialog

### PR DESCRIPTION
**Pull request by calmh**
Do we want something like this? It gets really long (folders come after the options)...

screen shot 2015-07-03 at 14 08 01

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
Looks good.

File: gui/scripts/syncthing/app.js [hunk 1]
```diff
@@ -11,7 +11,7 @@
 var syncthing = angular.module('syncthing', [
     'angularUtils.directives.dirPagination',
     'pascalprecht.translate',
-
+    'JSONedit',
```
Comment (AudriusButkevicius) [inline gui/scripts/syncthing/app.js]: Missing?

**Comment by uok** [comment]
warning note ... check
all config possibilities ... check
:+1: 

improvement idea: make groups (GUI, folder, etc) collapsible to reduce scrolling
also: is there a info in the docs with all default values where users can check if they want to reset something?

File: gui/scripts/syncthing/app.js (hunk 1 above)
Comment (Zillode) [inline gui/scripts/syncthing/app.js]: +1
Getting the following error https://docs.angularjs.org/error/$injector/modulerr?p0=syncthing&p1=Error:%20%5B$injector:modulerr

(like the idea tho)

**Comment by calmh** [comment]
The JSONedit stuff is a bug, it's a package I tried out that didn't work out - I'll clean it out. And see if I can implement what @uok suggests

**Comment by calmh** [comment]
BTW I didn't bother to implement an editor for arrays, so the listen address and discovery server options aren't present on this page. I don't think it's a big deal at the moment as they are in the normal config editor, someone else can implement that if they think it's important.

**Comment by calmh** [comment]
screen shot 2015-07-04 at 13 48 05

screen shot 2015-07-04 at 13 48 13

**Comment by uok** [comment]
nice! :smile:

**Pull request merged**